menus/menu_pool.py

```diff
@@ -204,10 +204,36 @@ def _build_nodes(self):
         return final_nodes
 
     def _mark_selected(self, nodes):
+        root_nodes = [node for node in nodes if not node.parent]
         for node in nodes:
-            node.selected = node.is_selected(self.request)
+            if node.is_selected(self.request):
+                node.selected = True
+                self._mark_ancestors(node)
+                self._mark_descendants(node)
+                self._mark_siblings(node, root_nodes)
+                return nodes
         return nodes
 
+    def _mark_ancestors(self, node):
```

Comment:
It'd be good to have some docstrings on these new functions, despite how simple they may be.